src/screens/json-rpc-sign-screen.tsx

```diff
@@ -65,11 +77,30 @@ export const JsonRpcSignScreen = () => {
           chainId,
         });
       }
-    } finally {
-      navigation.goBack();
     }
   }, [chainId, isAllowed, navigation, onPressReject, request, wallet]);
 
+  const checkContractAddress = useCallback(async () => {
+    if (isTransaction) {
+      const params = getTransactionFromJsonRpcRequest(request);
+
+      if (params?.from) {
+        const info = await verifyAddress(params.from);
+        info && setVerifyAddressResponse(info);
```

Comment:
Do not create boolean variable here
```suggestion
if (info) setVerifyAddressResponse(info);
```